[DTensor] Add `aten.view.dtype` op support

awgu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #144404

Fixes https://github.com/pytorch/pytorch/issues/144286

Viewing a tensor to a different dtype does not require any redistribution and can use the default strategy.

cc @H-Huang @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

Commit:
[DTensor] Add `aten.view.dtype` as pointwise op

[ghstack-poisoned]

Commit:
Update on "[DTensor] Add `aten.view.dtype` as pointwise op"


Fixes https://github.com/pytorch/pytorch/issues/144286

Viewing a tensor to a different dtype is pointwise and does not require any redistribution.

cc H-Huang kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

wanchaol commented on torch/distributed/tensor/_ops/_pointwise_ops.py:
```diff
@@ -398,6 +398,7 @@
     aten.trunc_.default,
     aten.where.self,
     aten.where.self_out,
+    aten.view.dtype,
```
should it be the `default_strategy` instead of the pointwise strategy? https://github.com/pytorch/pytorch/blob/main/torch/distributed/tensor/_ops/_tensor_ops.py#L41

pointwise strategy should also work but it would try handling broadcasting semantics which might not be necessary for `view.dtype`?

awgu replied:
oh oops, thanks for pointing this out! let me fix

awgu replied:
changed!

Commit:
Update on "[DTensor] Add `aten.view.dtype` as pointwise op"


Fixes https://github.com/pytorch/pytorch/issues/144286

Viewing a tensor to a different dtype is pointwise and does not require any redistribution.

cc H-Huang kwen2501 wanchaol fegin fduwjj wz337 wconstab d4l3k c-p-i-o

[ghstack-poisoned]

wanchaol reviewed (approved):
lgtm!

awgu commented:
@pytorchbot merge

awgu commented:
@pytorchbot merge
